pkg/chunk/chunk_store_utils.go

```diff
@@ -149,24 +194,40 @@ func ProcessCacheResponse(chunks []Chunk, keys []string, bufs [][]byte) (found [
 			missing = append(missing, chunks[i])
 			i++
 		} else if chunkKey > keys[j] {
-			level.Debug(util.Logger).Log("msg", "got chunk from cache we didn't ask for")
+			level.Error(util.Logger).Log("msg", "got chunk from cache we didn't ask for")
 			j++
 		} else {
-			chunk := chunks[i]
-			err = chunk.Decode(decodeContext, bufs[j])
-			if err != nil {
-				cacheCorrupt.Inc()
-				return
-			}
-			found = append(found, chunk)
+			requests = append(requests, decodeRequest{
+				chunk: chunks[i],
+				buf:   bufs[j],
```

Comment:
Where are you setting `decodeRequest.responses`?

Reply:
Good catch!